service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/GetAzureDiskAttachedVmStep.java

```diff
@@ -16,54 +15,58 @@
 
 /**
  * This steps works as a companion step for a disk deletion process. It gets identifier of a virtual
- * machine to which data disk is attached and saves it into flight's working map. This data is
- * required for 'undo' operation at DetachAzureDiskFromVmStep. It is required in case a disk has
- * been detached from a virtual machine, but we need to attach it back in corresponding 'undo'
- * operation.
+ * machine to which data disk is attached and saves it into flight's working map. It is required in
+ * case a disk has been detached from a virtual machine, but we need to attach it back in
+ * corresponding 'undo' operation. This is a separate step because the data is required for the
+ * 'undo' operation at DetachAzureDiskFromVmStep, and we need to guarantee its persistence - which
+ * we can't guarantee if it is from the 'do' portion of the same step.
  */
-public class GetAzureDiskAttachedVmStep implements DeleteControlledResourceStep {
+public class GetAzureDiskAttachedVmStep
```

Comment:
I don't have a better suggestion unfortunately, but just calling out that I find it rather confusing that this step is a `Get` step that also extends `DeleteAzureControlledResourceStep` and it has a `deleteResource` method that doesn't delete anything.

Reply:
I know. Naming things is hard.
There is a fairly extensive explanation in the comments, for this reason.